File: Source/src_libgdx/ProjectCube/src/com/cube/data/Animation.java
package com.cube.data;

import java.util.ArrayList;
import java.util.List;
import com.cube.common.ColorDepth;

public class Animation {
	private ColorDepth _depth;
	private List<Frame> _frames = new ArrayList<Frame>();

	private float _elapsedTime = 0;
	private int _currentFrameIndex = 0;
	private short _currentFrameDuration = 0;

	public Animation() {

	}

	public List<Frame> get_frames() {
		return _frames;
	}

	void set_frames(List<Frame> frames) {
		this._frames = frames;
		_currentFrameDuration = frames.get(0).get_duration();
	}

	ColorDepth get_depth() {
		return _depth;
	}

	void set_depth(ColorDepth depth) {
		this._depth = depth;
	}

	public void act(float delta) {
		_elapsedTime += delta * 1000;
		if (_elapsedTime >= _currentFrameDuration) {
			_elapsedTime = 0;
		

		_currentFrameIndex = _currentFrameIndex < (_frames.size() - 1) ? (_currentFrameIndex + 1) : 0;
		_currentFrameDuration = _frames.get(_currentFrameIndex).get_duration();
		}
	}

	public Frame getCurrentFrame() {
		return _frames.get(_currentFrameIndex);
	}
}

File: Source/src_spritekit/Project Cube/Project Cube/Animation.swift
//
//  Animation.swift
//  Project Cube
//
//  Created by Tamas Santa on 13/01/15.
//  Copyright (c) 2015 Tamas Santa. All rights reserved.
//

import Foundation

class Animation {
    
    var Depth: ColorDepth = ColorDepth.GrayScale
    var ElapsedTime: Float = 0
    var CurrentFrameIndex: Int = 0
    var CurrentFrameDuration: Int = 0
    
    var Frames: Array<Frame> {
        get {
            return _frames
        }
        set(frames)
        {
            _frames = frames
            if (_frames.first != nil)
            {
                CurrentFrameDuration = _frames.first!.Duration
            }
        }
    }
    private var _frames: Array<Frame> = []
    
    func Reset() {
        CurrentFrameIndex = 0;
        ElapsedTime = 0;
        
        if (_frames.first != nil)
        {
            CurrentFrameDuration = _frames.first!.Duration
        }
    }
    
    func Act(delta: Float) {
        ElapsedTime += delta;
        if ElapsedTime >= Float(CurrentFrameDuration * 500)
        {
            ElapsedTime = 0
            CurrentFrameIndex = CurrentFrameIndex < (_frames.count - 1) ? (CurrentFrameIndex + 1) : 0
            CurrentFrameDuration = _frames[CurrentFrameIndex].Duration
        }
    }
    
    func GetCurrentFrame() -> Frame {
        return _frames[CurrentFrameIndex]
    }
    
}